be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -68,20 +68,32 @@ Status JsonScanner::open() {
 StatusOr<ChunkPtr> JsonScanner::get_next() {
     SCOPED_RAW_TIMER(&_counter->total_ns);
     ChunkPtr src_chunk;
-    RETURN_IF_ERROR(_create_src_chunk(&src_chunk));
 
-    if (_cur_file_eof) {
-        RETURN_IF_ERROR(_open_next_reader());
-        _cur_file_eof = false;
-    }
-    Status status = _cur_file_reader->read_chunk(src_chunk.get(), _max_chunk_size, _src_slot_descriptors);
-    if (status.is_end_of_file()) {
-        _cur_file_eof = true;
-    }
+    // Read until we get a non-empty chunk.
+    do {
+        RETURN_IF_ERROR(_create_src_chunk(&src_chunk));
 
-    if (src_chunk->num_rows() == 0) {
-        return Status::EndOfFile("EOF of reading json file");
-    }
+        if (_cur_file_eof) {
+            // If all readers have been read, a EOF would be returned.
+            RETURN_IF_ERROR(_open_next_reader());
```

Comment:
if alll file are read, but the src_chunk is still zero. How to handle it?

Reply:
https://github.com/StarRocks/starrocks/blob/97b3c64a3ac8be25f60dc6a981cefc1822b9065e/be/src/exec/vectorized/json_scanner.cpp#L206-208
The `_open_next_reader` would return EOF and then the loop is end.